viewer/apiHunts.js

```diff
@@ -888,12 +889,24 @@ ${Config.arkimeWebURL()}sessions?expression=huntId==${huntId}&stopTime=${hunt.qu
         // clear out secret fields for users who don't have access to that hunt
         // if the user is not an admin and didn't create the hunt and isn't part of the user's list
         if (!req.user.hasRole('arkimeAdmin') && req.user.userId !== hunt.userId && hunt.users.indexOf(req.user.userId) < 0) {
-          // since hunt isn't cached we can just modify
-          hunt.id = '';
-          hunt.search = '';
-          hunt.userId = '';
-          hunt.searchType = '';
-          delete hunt.query;
+          let userHasRole = false;
+          if (hunt.roles.length) {
+            for (const role of hunt.roles) {
+              if (req.user.roles.indexOf(role) > -1) {
+                userHasRole = true;
+                break;
+              }
+            }
+          }
```

Comment:
Can this just be if hunt.roles.length && req.user.hasRole(hunt.roles) it already handles array vs single